media/windows/ECC-Ecobee/readme.md

```diff
@@ -0,0 +1,52 @@
+
+The data retrieved is extensive -- the Ecobee service provides detailed data covering over two dozen record types (even more for utility customers).  Some of this data is maintained on a historical basis, for our purposes here, primarily historical runtime information.  This data is reported from the thermostats every 15 minutes or so (more frequently on equipment status changes), and is maintained on intervals of every 5 minutes.  This historical data is quite large, with 288 records maintained for each thermostat for each day (12 5-min. increments * 24 hours).  So, retrieving 90 days of information will result in almost 26,000 records for one thermostat.
+
+Additional thermostat details are available as a snapshot in time for most of the remaining record types.  This includes point-in-time information on items such as current settings, notifications, programs, events, firmware version, etc.
+
+The application will first poll the historical runtime information for each thermostat.  The timeframe for each iteration is determined by examining the existing data from the SQLite database to determine the last revision interval (5-minute time slice) that was reported and written.  It will then poll the Ecobee service to determine the latest interval that was reported to the service.  If newer information is available, it will then be requested.  This process follows the recommendations given on the Ecobee site to avoid unnecessary polling requests (see [Thermostat Summary Polling](https://www.ecobee.com/home/developer/api/documentation/v1/operations/get-thermostat-summary.shtml) for more details).
+
+If no runtime information is currently available in the database, the application will default to using the first-reported date retrieved for this thermostat from the Ecobee service (essentially the in-service date).  Data requests for the runtime information only support up to 31 days retrieval per request; therefore, larger requests must be broken up into smaller ones not to exceed this limit.  The application automatically determines the retrieval window and will adjust the polling requests to accomodate this restriction.  This is generally only an issue during the first-run, when several months of data is to be retrieved for each thermostat, beginning with the in-service date.
+
+Following the polling of the runtime information, data is retrieved for the point-in-time or snapshot data.  One record is written for each data-type retrieved for each thermostat; so, the number of records written for each data-type is determined by the frequency by which the application is run.  That is, if it is run hourly, one record will be retrieved per thermostat, per data-type, for each hour.  Or, one per day, per thermostat, per data-type if run daily.
+## Quick Start
+In order to perform the initial setup for running the application:
+1. Install the Pyecobee library from the included archive file:  'pip install ./Pyecobee-mumblepins.zip -v'
```

Comment:
This zip is the mumblepins (maintained) version of the repository I used.  The only reason I included it in the master is because it was the version I used and I knew it worked with the code.  There really should be no reason it can’t be installed directly from his repository as an alternative...we’d just need to ensure any new releases were tested with my code to ensure they still work n the future.

Reply:
If mumblepins has a network-pip-installable version, that would be slightly better (i.e., `pip install ...something...` that pulls it down from the internet, vs. the local-install pip form you have).  But if not, we can just carry the zip here in git.

Minor ask: if we end up keeping the zip files, we probably don't need to maintain more than 1 version of it at the git head.  I.e., if v2 is replacing v1, there's no need for keeping v1 at the head (it'll be in the git history if needed).

Reply:
I’m not sure if it is directly-installable from the repository or not, since I’ve not attempted that.  I first downloaded the repository as a zip and installed from there.  Probably just reached the limit of my knowledge on this regarding using pip, though seems it should work to me.

And you’ve certainly have hit my knowledge limit re: git and not maintaining multiple versions at the head.  If you can give me some guidance on that I’ll take care of it...otherwise I’ll need to dig around a bit more ;).

And regarding multiple versions...since this posted Oct-1 release, I’ve made substantial code changes to continue to enhance the error checking and correction to handle more common error situations.  Haven’t posted them yet since I didn’t want to keep putting up minor release until you had time to look at moving it to production.  So, if you’re ready, I’ll issue another pull request for the very latest (well, it may take me till Mon-Tues, as we’re sill doing some Christmas family gatherings.). Just let me know about your availability.

Reply:
What I mean is can we `pip install mumblepins` (or some other name) that will pull down the code from the internet?  That's usually how things are installed via pip -- you don't usually have a local zip file to install from.  That being said, `pip install pyecobee` may download/install the "wrong" one (i.e., the old/unmaintained one -- not the mumblepins one).  See if the mumblepins guy has a pip entry.

As for what I meant by not having multiple copies in git is that in your 2nd commit, we can probably `git rm "ECC Pycobee Data v01.00.py"` so that the 2nd commit removes v1 and adds v2.  Make sense?

I'm actually off all this upcoming week -- want to get together to go through this stuff?  We can (finally!) deploy it on ECC resources, do all the pip/git stuff, ...etc.